### PR TITLE
Check isBulkCarry config  for nested relationships

### LINKED ISSUE #5205: Bulk carry forward configuration is not respected for fields in related tables
**Describe the bug**
The "Carry Forward" configuration for required fields in related tables seems to always be used instead of the "Bulk Carry Forward" configuration. If required fields are disabled in the "Carry Forward" configuration but are forcibly enabled in the "Bulk Carry Forward" configuration, they are not carried forward when you perform a bulk carry forward.

For fields/relationships that are required by default (such as prepType), an error message will be thrown before the new records can be created. For fields that are required by the user in schema config, new records will be created, but they cannot be saved until you input a value.

**To Reproduce**

1. Create a new Collection Object record
2. Fill in required fields and add a record in a related table (for example, Preparation)
3. Save the record
4. Go to the Form Meta using the gear icon. Click "Show Carry Forward button" and "Show Bulk Carry Forward count"
5. Next to "Show Carry Forward button," click the gear icon. In this dialog, find the table you added a record to under "Relationships". Click the gear icon next to it and uncheck a required field/relationship (for example, countAmt)
6. Close all dialogs to return to your Collection Object record.
7. Set the Carry Forward count to be greater than 1 and click "Carry Forward"
8. See error

**Expected behavior**
The "Bulk Carry Forward" configuration for fields in related tables should be respected when a bulk carry forward is performed.

**Screenshots**

https://github.com/user-attachments/assets/79d59525-70a0-4e2f-a05b-6ee4a3353df3

https://fwri1924-edge.test.specifysystems.org/specify/view/collectionobject/527506/?recordsetid=1933

Alternate behavior (prepType):

https://github.com/user-attachments/assets/3cc8be27-3b76-487f-9667-1aa0ca6c20b6

**Crash Report**
[Specify 7 Crash Report - 2024-08-07T18_25_51.891Z.txt](https://github.com/user-attachments/files/16532700/Specify.7.Crash.Report.-.2024-08-07T18_25_51.891Z.txt)

Please, also fill out the following information manually:

- OS: Mac
- Browser: Chrome
- Specify 7 Version: edge
- Database Name: fwri_1_9_24
- Collection name: Ichthyology
- User Name: spadmin

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #5205

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone

### Testing instructions

- verify that count field is required in schema config for prep table 
- Create a new CO record
- Click on the gear icon to access the form meta
- Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
- Click on the gear icon next to "Show Carry Forward Button"
- Uncheck a required field in a nested relationship, ex count in prep 
- Fill out required fields and save the record
- Input a value higher than 1 for "Bulk Carry Forward Count"
- Click "Carry Forward"
- [ ] verify the clone works and the amount of prep as been carried away for the bulk

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/FormMeta/CarryForward.tsx [hunk 1]
```diff
@@ -399,7 +399,9 @@ function CarryForwardCategory({
           const isUnique = uniqueFields.includes(field.name);
           const isRequired =
             isBulkConfig === true &&
-            (field.localization.isrequired || field.overrides.isRequired);
+            (field.localization.isrequired ||
+              field.overrides.isRequired ||
+              field.databaseColumn === 'CollectionObjectTypeID');
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/FormMeta/CarryForward.tsx:404]: hmm, are you sure that the bug in #5207 is affecting only this one field among all the fields in the database?


also, the condition here is only affecting the UI of this dialog - I don't think it has impact on the actual carry over operation, does it?

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]

File: specifyweb/frontend/js_src/lib/components/FormMeta/CarryForward.tsx (hunk 1 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/FormMeta/CarryForward.tsx:404]: yes it does; the COType is now being carried away when bulk carrying.
The issue here is that the COType wasn't required (so we could uncheck it) but we have now a rule that says that you always need a cotype and that this needs to be from the same treedef as the determination. Which was throwing the errors. 
Might have to do something similar in the next release for COGType

**Review by melton-jason — CHANGES_REQUESTED** [review]

**Review by emenslin — DISMISSED** [review]
### Testing instructions

- verify that count field is required in schema config for prep table 
- Create a new CO record
- Click on the gear icon to access the form meta
- Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
- Click on the gear icon next to "Show Carry Forward Button"
- Uncheck a required field in a nested relationship, ex count in prep 
- Fill out required fields and save the record
- Input a value higher than 1 for "Bulk Carry Forward Count"
- Click "Carry Forward"
- [ ] verify the clone works and the amount of prep as been carried away for the bulk 

When I uncheck the count field in preparations and try to carry forward I get an error 

https://github.com/user-attachments/assets/4ec838ff-9391-49b8-a9ca-3ae7d6a58185


This is what preparation fields in the bulk carry forward look like 
Screenshot 2024-08-13 112209

**Review by lexiclevenger — APPROVED** [review]
### Testing instructions

- verify that count field is required in schema config for prep table 
- Create a new CO record
- Click on the gear icon to access the form meta
- Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
- Click on the gear icon next to "Show Carry Forward Button"
- Uncheck a required field in a nested relationship, ex count in prep 
- Fill out required fields and save the record
- Input a value higher than 1 for "Bulk Carry Forward Count"
- Click "Carry Forward"
- [x] verify the clone works and the amount of prep as been carried away for the bulk 
---
I tested this with count and prepType on KUPaleoBot_2024_06_06 without issue:


https://github.com/user-attachments/assets/2cd893d6-f0af-4462-bd43-658bd0ee4433

**Review by combs-a — APPROVED** [review]
### Testing instructions

- verify that count field is required in schema config for prep table 
- Create a new CO record
- Click on the gear icon to access the form meta
- Enable the "Show Carry Forward Button" and "Show bulk carry forward count"
- Click on the gear icon next to "Show Carry Forward Button"
- Uncheck a required field in a nested relationship, ex count in prep 
- Fill out required fields and save the record
- Input a value higher than 1 for "Bulk Carry Forward Count"
- Click "Carry Forward"
- [X] verify the clone works and the amount of prep as been carried away for the bulk 

Was able to do it with PrepType, looking fine!

https://github.com/user-attachments/assets/e8fd9ec5-9362-42f0-9d75-e395b8dca60d

**Pull request merged**